code/modules/admin/verbs/adminhelp.dm

```diff
@@ -141,3 +141,234 @@ var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","
+	//		highlight special roles (0 = everyone has same looking name, 1 = antags / special roles get a golden name)
+/* Old ones that have Autoresponses.
+	var/mentor_msg = "<span class='notice'><b><font color=red>[selected_type]: </font>[get_options_bar(mob, 4, 1, 1, 0, ticket)] (<a href='?_src_=holder;take_ticket=\ref[ticket]'>[(ticket.status == TICKET_OPEN) ? "TAKE" : "JOIN"]</a>) (<a href='?src=\ref[usr];close_ticket=\ref[ticket]'>CLOSE</a>) (<a href='?_src_=holder;autoresponse=\ref[mob]'>AutoResponse...</a>):</b> [msg]</span>"
+	msg = "<span class='notice'><b><font color=red>[selected_type]: </font>[get_options_bar(mob, 2, 1, 1, 1, ticket)] (<a href='?_src_=holder;take_ticket=\ref[ticket]'>[(ticket.status == TICKET_OPEN) ? "TAKE" : "JOIN"]</a>) (<a href='?src=\ref[usr];close_ticket=\ref[ticket]'>CLOSE</a>) (<a href='?_src_=holder;autoresponse=\ref[mob]'>AutoResponse...</a>):</b> [msg]</span>"
+*/
+
+	var/mentor_msg = "<span class='notice'><b><font color=red>[selected_type]: </font>[get_options_bar(mob, 4, 1, 1, 0, ticket)] (<a href='?_src_=holder;take_ticket=\ref[ticket]'>[(ticket.status == TICKET_OPEN) ? "TAKE" : "JOIN"]</a>) (<a href='?src=\ref[usr];close_ticket=\ref[ticket]'>CLOSE</a>):</b> [msg]</span>"
+	msg = "<span class='notice'><b><font color=red>[selected_type]: </font>[get_options_bar(mob, 2, 1, 1, 1, ticket)] (<a href='?_src_=holder;take_ticket=\ref[ticket]'>[(ticket.status == TICKET_OPEN) ? "TAKE" : "JOIN"]</a>) (<a href='?src=\ref[usr];close_ticket=\ref[ticket]'>CLOSE</a>):</b> [msg]</span>"
+
+	var/admin_number_afk = 0
+
+	var/list/mentorholders = list()
+	var/list/debugholders = list()
+	var/list/modholders = list()
+	var/list/adminholders = list()
+	for(var/client/X in GLOB.admins)
+		if(R_MENTOR & X.holder.rights && !(R_ADMIN & X.holder.rights)) // we don't want to count admins twice. This list should be JUST mentors
+			mentorholders += X
+			if(X.is_afk())
+				admin_number_afk++
+		if(R_DEBUG & X.holder.rights || R_DEBUG & X.holder.rights) // Looking for anyone with +Debug which will be admins, developers, and developer mentors
+			debugholders += X
+			if(!(R_ADMIN & X.holder.rights))
+				if(X.is_afk())
+					admin_number_afk++
+
+		if(R_MOD & X.holder.rights || R_BAN & X.holder.rights) // Looking for anyone with +Ban which will be full mods and admins.
+			if(!(R_ADMIN & X.holder.rights))
+				modholders += X
+				if(X.is_afk())
+					admin_number_afk++
+		if(R_ADMIN & X.holder.rights || R_ADMIN & X.holder.rights) // just admins here please
+			adminholders += X
+			if(X.is_afk())
+				admin_number_afk++
+
+
+	switch(selected_type)
+		if("Gameplay/Job Inquiries")
+			if(mentorholders.len)
+				for(var/client/X in mentorholders) // Mentors get a message without buttons and no character name
+					if(X.get_preference_value(/datum/client_preference/staff/play_adminhelp_ping))
```

Comment:
it seems silly to have this check inside every part of the switch statement

Reply:
It's the only way it works, unfortunately. I've checked into it myself, and tested multiple times.